spec/support/schema_matcher.rb

```diff
@@ -0,0 +1,5 @@
+RSpec::Matchers.define :match_schema do |schema|
+  match do |metadata|
+    ValidateSchema.validate(metadata, schema)
+  end
+end
```

Comment:
Cool, excellent idea. What's the failure message when test fail for this matcher?

Reply:
It bubbles up the message from the `json-schema` gem. Unfortunately the message can sometimes be a little hard to follow, but it at the very least references the object it was trying to check and the property it failed on